### DISCUSSION TITLE
Import pkcs12 file into android

### DISCUSSION THREAD

**Discussion by robdyck** [strongswan/strongswan#1365] [Q&A] [answered]
I am trying to add a profile to android strongswan. The profile should use certificates for authentication. This requires importing a p12 file. I am using openssl tools for all the certificate related operations.

After creating the p12 I pushed it to the phone. From the "add profile" menu I touch "install user certificate". This opens my file manager and I select the p12 file. I am promped for a password. The p12 was produced without supplying an import password. I try simply OK. I am once again prompted for a password. This has worked for me in the past.

I decided to create a new p12 and I supplied a password. When repeating the steps and this time entering the password previously created, it continues to prompt for the password. I deliberately made the password short and simple and I am sure I entered it correctly.

I tried it on an older version of android that already had a working profile. I got the same problem.

I decided to try it the other way round. I pushed the older p12 to the newer phone. The older p12 had been produced without a password. I could now import this older p12. I was not prompted for password.

My conclusion is that p12's produced with a recent version of openssl are incompatible with strongswan on android. Is this an incorrect assumption or should I report it as a bug.

``openssl version
OpenSSL 3.0.5 5 Jul 2022 (Library: OpenSSL 3.0.5 5 Jul 2022)``

**Comment by tobiasbrunner** [comment, marked as answer]
You need to use the `-legacy` option when creating a PKCS#12 container with OpenSSL 3.x, see the documentation [here](https://docs.strongswan.org/docs/6.0/pki/pkiQuickstart.html#_pkcs12_container).

**Reply by robdyck** [reply]
Some times using your old notes doesn't work out. Thanks for the guidance.

**Comment by mrbluecoat** [comment]
Is this still required? I heard **3DES** (used by `-legacy`) [will be prohibited at the end of 2023](https://www.encryptionconsulting.com/why-3des-or-triple-des-is-officially-being-retired/).
